dtm0/domain.c

```diff
@@ -99,18 +106,30 @@ static int dtm0_domain_level_enter(struct m0_module *module)
 	enum dtm0_domain_level     level = module->m_cur + 1;
 	struct m0_dtm0_domain     *dod = dtm0_module2domain(module);
 	struct m0_dtm0_domain_cfg *cfg = &dod->dod_cfg;
+	struct m0_reqh            *reqh = cfg->dod_reqh;
 
 	M0_ENTRY("dod=%p level=%d level_name=%s",
 	         dod, level, levels_dtm0_domain[level].ml_name);
 	switch (level) {
 	case M0_DTM0_DOMAIN_LEVEL_INIT:
+		if (reqh != NULL && reqh->rh_beseg != (void *)1 &&
```

Comment:
In Non-UT code, we should not have such invalid & fake pointer.

Reply:
I think motr client init function uses (void *) to initialize params of request handle, i don't seen any other approach we can use here.